fontbe/src/hmtx.rs

```diff
@@ -0,0 +1,80 @@
+//! Generates a [hmtx](https://learn.microsoft.com/en-us/typography/opentype/spec/hmtx) table.
+
+use fontdrasil::orchestration::Work;
+use write_fonts::{
+    dump_table,
+    tables::{hmtx::Hmtx, vmtx::LongMetric},
+};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Context},
+};
+
+pub struct RawHmtx {
+    pub buf: Vec<u8>,
+}
+
+struct HmtxWork {}
+
+pub fn create_hmtx_work() -> Box<BeWork> {
+    Box::new(HmtxWork {})
+}
+
+impl Work<Context, Error> for HmtxWork {
+    /// Generate [hmtx](https://learn.microsoft.com/en-us/typography/opentype/spec/hmtx)
+    fn exec(&self, context: &Context) -> Result<(), Error> {
+        let static_metadata = context.ir.get_final_static_metadata();
+        let var_model = &static_metadata.variation_model;
+        let default_location = var_model.default_location();
+
+        let mut long_metrics: Vec<LongMetric> = static_metadata.glyph_order.iter()
+            .map(|gn| {
+                let bbox = context.get_glyph(gn).bbox();
+                let ir_glyph = context.ir.get_glyph_ir(gn);
+                let Some(ir_instance) = ir_glyph.sources.get(default_location) else {
+                    panic!("{gn} is not defined at the default location. Unnecessary limitation if it covers the default.");
+                };
+                let advance: u16 = ir_instance.width as u16;
+
+                LongMetric {
+                    advance,
+                    side_bearing: bbox.x_min
```

Comment:
Source can contain float advance widths as well as float coordinates. Where does the rounding from floats to int happen? in the IR or here in the BE?
To match ufo2ft we want to use [otRound](https://github.com/fonttools/fonttools/blob/444a349ef4dd1bae2d43f6ea0e06d73fb6466b9a/Lib/fontTools/misc/roundTools.py#L23) for this.

Reply:
The IR GlyphInstance has `width: f64,` and I'm using as u16 here to convert it.

Reply:
I think that will make a very nice discrete followon; tracking in https://github.com/googlefonts/fontmake-rs/issues/147.